internal/controller/template_controller_test.go

```diff
@@ -149,44 +206,57 @@ func applyYAMLFilesFromDirectory(ctx context.Context, dir string) {
 			}
 			Expect(err).NotTo(HaveOccurred())
 		}
-	}
+
+		// Entry is a directory - Continue traversal
+		return nil
+	})
+	Expect(err).NotTo(HaveOccurred())
+}
+
+// Waits until the given K8s Client resource is deleted, or times out
+func waitForDeletion(ctx context.Context, obj *unstructured.Unstructured, timeout, interval time.Duration) {
+	fmt.Printf("Waiting for deletion of resource %s/%s\n", obj.GetKind(), obj.GetName())
+	Eventually(func() error {
+		err := k8sClient.Get(ctx, client.ObjectKeyFromObject(obj), obj)
+		if errors.IsNotFound(err) {
+			fmt.Printf("Resource %s/%s deleted done\n", obj.GetKind(), obj.GetName())
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("Resource %s/%s not deleted yet\n", obj.GetKind(), obj.GetName())
+		return fmt.Errorf("Resource %s/%s still exists", obj.GetKind(), obj.GetName())
+	}, timeout, interval).Should(Succeed())
 }
 
+// Delete all YAML files defined in the given directory, dir, as well as all sub-directories
 func deleteYAMLFilesFromDirectory(ctx context.Context, dir string) {
```

Comment:
Agreed - This will be renamed in my next commit! This was trickier to do than I thought, but I settled on renaming the Apply and Delete functions to `applyResourcesFromYAMLInDir` and `deleteResourcesFromYAMLInDir`. Are these names more clear?